schemas/iso19139/src/main/plugin/iso19139/extract-relations.xsl

```diff
@@ -126,8 +127,9 @@
               <xsl:value-of select="concat('#', upper-case(util:twoCharLangCode($lang, 'EN')))"/>
             </xsl:variable>
             <xsl:variable name="url" select="gmd:linkage/gmd:URL"/>
+            <xsl:variable name="uid" select="uuid:randomUUID()"/>
             <id>
-              <xsl:value-of select="$url"/>
+              <xsl:value-of select="$uid"/>
```

Comment:
This makes sense, thank you for the improvement

Reply:
Perhaps matching based on URL + a counter be enough? This id does not have to last very long ... only long enough for the remove to function.